src/Reflection/ReflectionClass.php

```diff
@@ -236,7 +238,16 @@ public static function createFromNode(
      */
     public function getShortName() : string
     {
-        return $this->node->name;
+        if (!$this->isAnonymous()) {
+            return $this->node->name;
+        }
+
+        $fileName = $this->getFileName();
+        if ($fileName === null) {
```

Comment:
> It can be probably solved but I have to figure out how to solve start and end line numbers. I think it would be better to do it in another PR.

Hardcoding them would be OK, as well as using the file contents hash as the file name.